services/spar/src/Spar/Data.hs

```diff
@@ -619,15 +623,36 @@ lookupScimToken ::
   ScimToken ->
   m (Maybe ScimTokenInfo)
 lookupScimToken token = do
-  mbRow <- retry x1 . query1 sel $ params Quorum (Identity token)
-  pure $ fmap fromScimTokenRow mbRow
-  where
-    sel :: PrepQuery R (Identity ScimToken) ScimTokenRow
+  let tokenHash = hashScimToken token
+  mbRow <- retry x1 . query1 sel $ params Quorum (tokenHash, token)
```

Comment:
This can return both hashed and unhashed in any order, if both are still in the DB (race condition).

Reply:
done